dbms/src/Storages/PathCapacityMetrics.cpp

```diff
@@ -187,33 +235,45 @@ FsStats PathCapacityMetrics::getFsStats(bool finalize_capacity)
         total_stat.capacity_size = capacity_quota;
         total_stat.avail_size = std::min(total_stat.avail_size, total_stat.capacity_size - total_stat.used_size);
     }
-
     // PD get weird if used_size == 0, make it 1 byte at least
     total_stat.used_size = std::max<UInt64>(1, total_stat.used_size);
 
-    const double avail_rate = 1.0 * total_stat.avail_size / total_stat.capacity_size;
-    // Default threshold "schedule.low-space-ratio" in PD is 0.8, log warning message if avail ratio is low.
-    if (avail_rate <= 0.2)
-        LOG_WARNING(
-            log,
-            "Available space is only {:.2f}% of capacity size. Avail size: {}, used size: {}, capacity size: {}",
-            avail_rate * 100.0,
-            formatReadableSizeWithBinarySuffix(total_stat.avail_size),
-            formatReadableSizeWithBinarySuffix(total_stat.used_size),
-            formatReadableSizeWithBinarySuffix(total_stat.capacity_size));
-    total_stat.ok = 1;
-
+    // Just report local disk capacity and available size is enough
     CurrentMetrics::set(CurrentMetrics::StoreSizeCapacity, total_stat.capacity_size);
     CurrentMetrics::set(CurrentMetrics::StoreSizeAvailable, total_stat.avail_size);
     CurrentMetrics::set(CurrentMetrics::StoreSizeUsed, total_stat.used_size);
 
+    size_t remote_used_size = 0;
     if (finalize_capacity && S3::ClientFactory::instance().isEnabled())
     {
+        {
+            std::unique_lock<std::mutex> lock(mutex);
+            for (const auto & [keyspace_id, used_bytes] : keyspace_id_to_used_bytes)
+            {
+                UNUSED(keyspace_id);
+                total_stat.used_size += used_bytes;
+                remote_used_size += used_bytes;
+            }
+        }
+
         // When S3 is enabled, use a large fake stat to avoid disk limitation by PD.
         // EiB is not supported by TiUP now. https://github.com/pingcap/tiup/issues/2139
         total_stat.capacity_size = 1024UL * 1024UL * 1024UL * 1024UL * 1024UL; // 1PB
         total_stat.avail_size = total_stat.capacity_size - total_stat.used_size;
     }
+    CurrentMetrics::set(CurrentMetrics::StoreSizeUsedRemote, remote_used_size);
+
+    const double avail_rate = 1.0 * total_stat.avail_size / total_stat.capacity_size;
+    // Default threshold "schedule.low-space-ratio" in PD is 0.8, log warning message if avail ratio is low.
+    if (avail_rate <= 0.2)
+        LOG_WARNING(
+            log,
+            "Available space is only {:.2f}% of capacity size. Avail size: {}, used size: {}, capacity size: {}",
+            avail_rate * 100.0,
+            formatReadableSizeWithBinarySuffix(total_stat.avail_size),
+            formatReadableSizeWithBinarySuffix(total_stat.used_size),
+            formatReadableSizeWithBinarySuffix(total_stat.capacity_size));
```

Comment:
I think this warning logging is more useful for "local disk" available rate is low. Move it back to before line 241?

Reply:
Fixed.